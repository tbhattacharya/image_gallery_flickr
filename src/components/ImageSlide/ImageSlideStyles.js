export const styles = {
    imageView: {
        display: 'flex',
        justifyContent: 'center',
        marginTop: '20px'
    },
    image: {
        width: '70%',
        height: '70%'
    },
    imageFooter: {
        display: 'flex',
        justifyContent: 'center',
        marginTop: '10px',
        fontSize: '14px'
    },
    imageFooterContent: {
        width: '40%',
        height: '20%',
        color: 'wheat'
    },
    imageName:{
        fontSize: '18px'
    },
    imageBy: {
        margin: '10px 30px'
    }
}