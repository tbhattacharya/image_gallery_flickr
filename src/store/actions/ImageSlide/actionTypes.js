export const FETCH_IMAGE_INFO_SUCCESS = 'FETCH_IMAGE_INFO_SUCCESS';
export const FETCH_IMAGE_INFO_REQ = 'FETCH_IMAGE_INFO_REQ';
export const CLEAR_IMAGE_INFO_REQ = 'CLEAR_IMAGE_INFO_REQ';
export const CLEAR_IMAGE_INFO = 'CLEAR_IMAGE_INFO';