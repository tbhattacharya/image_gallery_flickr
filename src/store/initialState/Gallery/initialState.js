export const initialState = {
    photoGrid: [],
    galleryInfo: {}
}