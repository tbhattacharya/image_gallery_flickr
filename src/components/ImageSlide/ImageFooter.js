import React from 'react';
import { truncate } from 'lodash';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { withStyles } from '@material-ui/core/styles';
import { styles } from './ImageSlideStyles';
import Grid from '@material-ui/core/Grid';
import moment from 'moment';

class ImageFooter extends React.Component {

    render() {
        const { classes, imageInfo } = this.props;
        //TODO - Create the more link for the description
        const description = (imageInfo && imageInfo.description) ? imageInfo.description._content.replace(/\n/g, '<br />') : '';
        return (
            (imageInfo.id ? (
                <div className={classes.imageFooter}>
                    <div className={classes.imageFooterContent}>
                        <Grid container spacing={24}>
                            <Grid item xs={6} >
                                <div className={classes.imageName}>
                                    {imageInfo.title._content}
                                </div>
                                <div className={classes.imageBy}>
                                    Image By {imageInfo.owner.realname}
                                </div>
                                <div className={classes.imageBy}>
                                    Dated {moment(imageInfo.dates.taken).format('DD MMM YYYY')}
                                </div>
                            </Grid>
                            <Grid item xs={6} >
                                <div dangerouslySetInnerHTML={{ __html: truncate(description,{length: 120}) }} style={{maxHeight: '10px'}}></div>
                        </Grid>
                        </Grid>
                </div></div >) : '')
        );
    }
}

const mapStateToProps = state => createStructuredSelector(
    {
        imageInfo: state => state.imagedata.imageInfo
    });


export default withStyles(styles)(connect(mapStateToProps, null)(ImageFooter));