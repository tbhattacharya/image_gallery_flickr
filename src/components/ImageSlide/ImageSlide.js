import React from 'react';
import { isEmpty, forEach } from 'lodash';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect'
import * as actionType from '../../store/actions/ImageSlide/actionTypes';
import { withStyles } from '@material-ui/core/styles';
import { styles } from './ImageSlideStyles';
import { LazyLoadImage } from 'react-lazy-load-image-component';
import { Thumbs } from '../../utils/Constants';
import 'react-lazy-load-image-component/src/effects/blur.css';
import ImageFooter from './ImageFooter';

class ImageSlide extends React.Component {

    constructor(props) {
        super(props);
        const param = this.props.location.state;
        this.state = {
            id: param.id,
            secret: param.secret
        }
    }

    componentDidMount() {
        if (isEmpty(this.props.imageInfo)) {
            this.props.fetchPhotoInformation(this.state.id, this.state.secret);
        }
    }

    componentWillUnmount() {
        //Clear the redux store for image data so that new data is loaded when the page is revisited again
        this.props.clearPhotoInformation();
    }

    render() {
        const { classes, imageInfo } = this.props;
        //Create image url for Lazy loader
        const src = `https://farm${imageInfo.farm}.staticflickr.com/${imageInfo.server}/${imageInfo.id}_${imageInfo.secret}.jpg`;
        return (
            <div>
                <div className={classes.imageView}>
                    <div className={classes.image}>
                        {imageInfo.id ?
                            (<LazyLoadImage
                                height={Thumbs.height}
                                src={src}
                                effect="blur"
                                width={Thumbs.width} />) : null}
                    </div>
                </div>
                <ImageFooter />
            </div>
        );
    }
}

/** 
 * Use createStructuredSelector to improve performance, since memoized selectors cut down on the number of new
 * object/array references being returned from mapState functions.
*/
const mapStateToProps = state => createStructuredSelector(
    {
        imageInfo: state => state.imagedata.imageInfo
    });

const mapDispatchToProps = dispatch => {
    return {
        fetchPhotoInformation: (id, secret) => dispatch({ type: actionType.FETCH_IMAGE_INFO_REQ, param: { id: id, secret: secret } }),
        clearPhotoInformation: () => dispatch({ type: actionType.CLEAR_IMAGE_INFO_REQ })
    }
}

export default withStyles(styles)(connect(mapStateToProps, mapDispatchToProps)(ImageSlide));