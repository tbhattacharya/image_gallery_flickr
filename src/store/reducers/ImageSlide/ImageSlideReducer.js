import { initialState } from "../../initialState/ImageSlide/initialState";
import * as actionType from '../../actions/ImageSlide/actionTypes';

export const ImageSlideReducer = (state = initialState, action) => {
    switch (action.type) {
        case actionType.FETCH_IMAGE_INFO_SUCCESS:
            return {
                ...state,
                imageInfo: action.imageInfo
            }
        case actionType.CLEAR_IMAGE_INFO:
            return initialState;
        default:
            return state;
    }
}