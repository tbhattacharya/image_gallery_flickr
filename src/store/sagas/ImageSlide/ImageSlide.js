import * as actionType from '../../actions/ImageSlide/actionTypes';
import { call, put, takeLatest, all } from 'redux-saga/effects';
import { fetchImageInformationApi } from '../../../service/HttpService';

function* fetchImageInfo({ param }) {
    try {
        //Fetch image details from service
        const { data } = yield call(fetchImageInformationApi, param.id, param.secret);
        if (data) {
            //Only image related information is needed, discard the rest
            yield put({ type: actionType.FETCH_IMAGE_INFO_SUCCESS, imageInfo: data.photo });
        }
    } catch (e) {
        console.log('Exception in fetching Image information', e);
    }
}

function* clearImageInfo() {
    try {
        yield put({ type: actionType.CLEAR_IMAGE_INFO });
    } catch (e) {
        console.log('Exception in clearing Image information', e);
    }
}

export function* watchImgageInfo() {
    try {
        yield all([
            takeLatest(actionType.FETCH_IMAGE_INFO_REQ, fetchImageInfo),
            takeLatest(actionType.CLEAR_IMAGE_INFO_REQ, clearImageInfo)
        ]);
    } catch (e) {
        console.log('Exception in Imgae info saga watcher ', e);
    }
}