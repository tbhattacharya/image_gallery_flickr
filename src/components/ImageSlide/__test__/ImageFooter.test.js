import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import { withStyles } from '@material-ui/core/styles';
import configureStore from 'redux-mock-store'
import { initialState } from '../../../store/initialState/ImageSlide/initialState';
import { image } from '../../../utils/test-utils/GalleryData';
import ImageFooter from '../ImageFooter';
const mockStore = configureStore()


describe("Testing Image Footer", function () {
  const store = mockStore(initialState)
  const shallow = createShallow();
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ImageFooter store={store} imageInfo= {image}/>)
  })

  it("Expect Image Footer to be drawn without exception", function () {
    expect(wrapper).toBeDefined();
  });

  it("Expect Image Footer to match snapshot", function () {
    expect(wrapper).toMatchSnapshot();
  });
});