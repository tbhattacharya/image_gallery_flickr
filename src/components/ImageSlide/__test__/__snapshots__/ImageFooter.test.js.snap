// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Image Footer Expect Image Footer to match snapshot 1`] = `
<Connect(ImageFooter)
  classes={
    Object {
      "image": "Connect-ImageFooter--image-02",
      "imageBy": "Connect-ImageFooter--imageBy-06",
      "imageFooter": "Connect-ImageFooter--imageFooter-03",
      "imageFooterContent": "Connect-ImageFooter--imageFooterContent-04",
      "imageName": "Connect-ImageFooter--imageName-05",
      "imageView": "Connect-ImageFooter--imageView-01",
    }
  }
  imageInfo={
    Object {
      "comments": Object {
        "_content": "612",
      },
      "dates": Object {
        "lastupdate": "1537079823",
        "posted": "1454423486",
        "taken": "2016-01-05 09:13:59",
        "takengranularity": "0",
        "takenunknown": "0",
      },
      "dateuploaded": "1454423486",
      "description": Object {
        "_content": "Haukilahti, Espoo, Finland
    
    Shot with Panasonic DMC-LX100, processed with Lightroom CC 2015.",
      },
      "editability": Object {
        "canaddmeta": 0,
        "cancomment": 0,
      },
      "farm": 2,
      "geoperms": Object {
        "iscontact": 0,
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
      },
      "id": "24679948091",
      "isfavorite": 0,
      "license": "0",
      "location": Object {
        "accuracy": "16",
        "context": "0",
        "country": Object {
          "_content": "Finland",
          "place_id": "5ryEXkVTUb66Tnx33g",
          "woeid": "23424812",
        },
        "county": Object {
          "_content": "Helsingin",
          "place_id": "paE.WWFUVLvDkOxqOA",
          "woeid": "56120851",
        },
        "latitude": "60.154233",
        "locality": Object {
          "_content": "Espoo",
          "place_id": "0dYvnjJUVL7l.nI",
          "woeid": "564617",
        },
        "longitude": "24.776336",
        "place_id": "0dYvnjJUVL7l.nI",
        "region": Object {
          "_content": "Uusimaa",
          "place_id": "_7C9msNTW7n35YTFQQ",
          "woeid": "29389247",
        },
        "woeid": "564617",
      },
      "media": "photo",
      "notes": Object {
        "note": Array [
          Object {
            "_content": "winter",
            "author": "62118446@N03",
            "authorisdeleted": 0,
            "authorispro": 1,
            "authorname": "cbrutel",
            "authorrealname": "Christophe Brutel",
            "h": "27",
            "id": "72157664079284881",
            "photo_id": "24679948091",
            "w": "27",
            "x": "212",
            "y": "8",
          },
          Object {
            "_content": "swan",
            "author": "62118446@N03",
            "authorisdeleted": 0,
            "authorispro": 1,
            "authorname": "cbrutel",
            "authorrealname": "Christophe Brutel",
            "h": "27",
            "id": "72157663534634689",
            "photo_id": "24679948091",
            "w": "27",
            "x": "262",
            "y": "8",
          },
          Object {
            "_content": "Finland",
            "author": "62118446@N03",
            "authorisdeleted": 0,
            "authorispro": 1,
            "authorname": "cbrutel",
            "authorrealname": "Christophe Brutel",
            "h": "27",
            "id": "72157664047992672",
            "photo_id": "24679948091",
            "w": "27",
            "x": "262",
            "y": "40",
          },
          Object {
            "_content": "Espoo",
            "author": "62118446@N03",
            "authorisdeleted": 0,
            "authorispro": 1,
            "authorname": "cbrutel",
            "authorrealname": "Christophe Brutel",
            "h": "27",
            "id": "72157661906899043",
            "photo_id": "24679948091",
            "w": "27",
            "x": "212",
            "y": "40",
          },
          Object {
            "_content": "ice",
            "author": "62118446@N03",
            "authorisdeleted": 0,
            "authorispro": 1,
            "authorname": "cbrutel",
            "authorrealname": "Christophe Brutel",
            "h": "27",
            "id": "72157664176344186",
            "photo_id": "24679948091",
            "w": "27",
            "x": "212",
            "y": "70",
          },
          Object {
            "_content": "Baltic",
            "author": "62118446@N03",
            "authorisdeleted": 0,
            "authorispro": 1,
            "authorname": "cbrutel",
            "authorrealname": "Christophe Brutel",
            "h": "27",
            "id": "72157664079307221",
            "photo_id": "24679948091",
            "w": "27",
            "x": "262",
            "y": "70",
          },
        ],
      },
      "owner": Object {
        "iconfarm": 8,
        "iconserver": "7226",
        "location": "Espoo, Finland",
        "nsid": "62118446@N03",
        "path_alias": "christophebrutel",
        "realname": "Christophe Brutel",
        "username": "cbrutel",
      },
      "people": Object {
        "haspeople": 0,
      },
      "publiceditability": Object {
        "canaddmeta": 0,
        "cancomment": 1,
      },
      "rotation": 0,
      "safety_level": "0",
      "secret": "dd3e3f117f",
      "server": "1671",
      "tags": Object {
        "tag": Array [
          Object {
            "_content": "aube",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-267406",
            "machine_tag": false,
            "raw": "Aube",
          },
          Object {
            "_content": "baltic",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-9837",
            "machine_tag": false,
            "raw": "Baltic",
          },
          Object {
            "_content": "baltique",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-433049",
            "machine_tag": false,
            "raw": "Baltique",
          },
          Object {
            "_content": "bird",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-594",
            "machine_tag": 0,
            "raw": "Bird",
          },
          Object {
            "_content": "brume",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-87231",
            "machine_tag": 0,
            "raw": "Brume",
          },
          Object {
            "_content": "cygne",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-64137",
            "machine_tag": 0,
            "raw": "Cygne",
          },
          Object {
            "_content": "cygnetubercule",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-45578685",
            "machine_tag": 0,
            "raw": "Cygne tubercule",
          },
          Object {
            "_content": "dawn",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-1998",
            "machine_tag": false,
            "raw": "Dawn",
          },
          Object {
            "_content": "espoo",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-1430",
            "machine_tag": false,
            "raw": "Espoo",
          },
          Object {
            "_content": "finland",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-1436",
            "machine_tag": false,
            "raw": "Finland",
          },
          Object {
            "_content": "frost",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-3342",
            "machine_tag": 0,
            "raw": "Frost",
          },
          Object {
            "_content": "frozen",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-2284",
            "machine_tag": 0,
            "raw": "Frozen",
          },
          Object {
            "_content": "gel",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-50693",
            "machine_tag": 0,
            "raw": "Gel",
          },
          Object {
            "_content": "glace",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-96972",
            "machine_tag": false,
            "raw": "Glace",
          },
          Object {
            "_content": "haukilahti",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-2302284",
            "machine_tag": false,
            "raw": "Haukilahti",
          },
          Object {
            "_content": "hiver",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-6074",
            "machine_tag": false,
            "raw": "Hiver",
          },
          Object {
            "_content": "ice",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-682",
            "machine_tag": false,
            "raw": "Ice",
          },
          Object {
            "_content": "leica",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-96",
            "machine_tag": false,
            "raw": "Leica",
          },
          Object {
            "_content": "leverdesoleil",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-134742",
            "machine_tag": false,
            "raw": "Lever de Soleil",
          },
          Object {
            "_content": "mer",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-381",
            "machine_tag": false,
            "raw": "Mer",
          },
          Object {
            "_content": "mist",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-398",
            "machine_tag": 0,
            "raw": "Mist",
          },
          Object {
            "_content": "muteswan",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-155961",
            "machine_tag": 0,
            "raw": "Mute swan",
          },
          Object {
            "_content": "oiseau",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-52410",
            "machine_tag": 0,
            "raw": "Oiseau",
          },
          Object {
            "_content": "panasonic",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-11243",
            "machine_tag": false,
            "raw": "Panasonic",
          },
          Object {
            "_content": "sea",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-228",
            "machine_tag": false,
            "raw": "Sea",
          },
          Object {
            "_content": "sunrise",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-1370",
            "machine_tag": false,
            "raw": "Sunrise",
          },
          Object {
            "_content": "swan",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-2752",
            "machine_tag": 0,
            "raw": "Swan",
          },
          Object {
            "_content": "winter",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-201",
            "machine_tag": false,
            "raw": "Winter",
          },
          Object {
            "_content": "lx100",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-9993892",
            "machine_tag": false,
            "raw": "lx100",
          },
          Object {
            "_content": "2016",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-392175",
            "machine_tag": false,
            "raw": "2016",
          },
          Object {
            "_content": "january",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-7328",
            "machine_tag": false,
            "raw": "january",
          },
          Object {
            "_content": "travel",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-121",
            "machine_tag": false,
            "raw": "travel",
          },
          Object {
            "_content": "nature",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-791",
            "machine_tag": false,
            "raw": "nature",
          },
          Object {
            "_content": "europe",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-3272",
            "machine_tag": false,
            "raw": "europe",
          },
          Object {
            "_content": "color",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-544",
            "machine_tag": false,
            "raw": "color",
          },
          Object {
            "_content": "colour",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-545",
            "machine_tag": false,
            "raw": "colour",
          },
          Object {
            "_content": "photo",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-2326",
            "machine_tag": false,
            "raw": "photo",
          },
          Object {
            "_content": "photos",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-4634",
            "machine_tag": false,
            "raw": "photos",
          },
          Object {
            "_content": "photography",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-1935",
            "machine_tag": false,
            "raw": "photography",
          },
          Object {
            "_content": "raw",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-3444",
            "machine_tag": false,
            "raw": "raw",
          },
          Object {
            "_content": "landscape",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-1695",
            "machine_tag": 0,
            "raw": "landscape",
          },
          Object {
            "_content": "seascape",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-441",
            "machine_tag": false,
            "raw": "seascape",
          },
          Object {
            "_content": "waterscape",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-168898",
            "machine_tag": false,
            "raw": "waterscape",
          },
          Object {
            "_content": "water",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-800",
            "machine_tag": false,
            "raw": "water",
          },
          Object {
            "_content": "outdoor",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-4890",
            "machine_tag": false,
            "raw": "outdoor",
          },
          Object {
            "_content": "explored",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-2077205",
            "machine_tag": false,
            "raw": "explored",
          },
          Object {
            "_content": "explore",
            "author": "62118446@N03",
            "authorname": "cbrutel",
            "id": "62095392-24679948091-11820",
            "machine_tag": false,
            "raw": "explore",
          },
        ],
      },
      "title": Object {
        "_content": "Braving the cold",
      },
      "urls": Object {
        "url": Array [
          Object {
            "_content": "https://www.flickr.com/photos/christophebrutel/24679948091/",
            "type": "photopage",
          },
        ],
      },
      "usage": Object {
        "canblog": 0,
        "candownload": 0,
        "canprint": 0,
        "canshare": 0,
      },
      "views": "4429063",
      "visibility": Object {
        "isfamily": 0,
        "isfriend": 0,
        "ispublic": 1,
      },
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
